lib/pbench/server/api/__init__.py

```diff
@@ -76,12 +76,14 @@ def register_endpoints(api, app, config):
         resource_class_args=(config, logger),
     )
     api.add_resource(
-        IndexMappings,
-        f"{base_uri}/index/mappings/<string:index_key>",
+        DatasetMappings,
+        f"{base_uri}/dataset/mappings/<string:dataset_view>",
         resource_class_args=(config, logger),
     )
     api.add_resource(
-        IndexSearch, f"{base_uri}/index/search", resource_class_args=(config, logger),
+        DatasetSummary,
+        f"{base_uri}/dataset/summary",
```

Comment:
Hmm.

In terms of "view" mapping, yes, I'd thought maybe "run-data" is something like a "summary" vs the various details of "result-data-sample" and "result-data".

However, I think the basic "summary" view is `datasets/detail`, which shows an entire "run-data" document. And we have two separate "search" APIs for "run-data" -- `datasets/list` (which currently selects by user, time range, and controller but is moving to "metadata" constraint) and `index/search` which selects by document fields.

Maybe `datasets/search` would be good. And I think that would merge the old `index/search` and your new `datasets/values`, because they're all returning a set of documents based on document field value matches (derived from the namespace queries, which could also maybe be merged).

Reply:
> Maybe `datasets/search` would be good.

We have an existing branch in the URI "tree" labeled "dataset", singular.  I recommend against creating another branch labeled "datasets" (plural), as I don't think there is enough distinction between the two labels.

Reply:
Technically, we have an existing "branch" labeled "datasets" ... it's the addition of the singular that's inconsistent; but thanks for noticing that. 😄  Nikhil should fix that!